python/dgl/optim/pytorch/sparse_optim.py

```diff
@@ -74,7 +73,7 @@ def step(self):
         if self._first_step:
             for emb in self._params:
                 for _, data in emb._trace:
-                    if data.grad.data.device.type == "cuda":
+                    if data.grad.device.type == "cuda":
```

Comment:
Is this change intended?

Reply:
Yes. `data.grad` is a tensor, and `data.grad.device` is enough to get its device. `tensor.data` is an outdated grammar in PyTorch.

If you look at the following lines 80-83, they all use `data.grad.device`.